### ISSUE TITLE
gradle plugin as contribution

### ISSUE THREAD

**Issue by xenoterracide** [spotbugs/spotbugs-gradle-plugin#461]
I hate how the existing plugin works, largely because it doesn't actually support outputting the report as text for CI, it's a file no matter what you do. I looked into it, it's just executing the Jar, instead of calling API's, I have to imagine this is also slightly less performant (calling exec, memory etc). if I made a plugin that worked more like the maven plugin, would spotbugs consider bringing it under the umbrella? I kind of don't want to be the only maintainer. Also, would someone be willing to give me pointers on the spotbugs API's I need to be using?

**Comment by jscancella** [comment]
Why do you need it to output the report for CI? Usually the report file is uploaded to some other program to track and display the issues found.
I would have to defer to @KengoTODA on if we would accept it, my guess though would be no as this one suits our needs just fine. If you need help with the spotbugs API I would look at the documentation in the https://github.com/spotbugs/spotbugs repo.

**Comment by xenoterracide** [comment]
I hate not just being able to see the needed output as plain text on the console. I filed [this issue](https://github.com/spotbugs/spotbugs-gradle-plugin/issues/172) here years ago and it hasn't been fixed which is why I was asking about another plugin on spotbugs/spotbugs  because the documentation says this is still considered an external plugin. I'll go ahead and close this issue, it's obvious that it's never going to be fixed.

**Comment by jscancella** [comment]
You are free to submit a pull request. A lot of us are not paid to work on this, and do it with our free time and out of the goodness of our heart. There is a current work around so it isn't high priority. It doesn't mean we will never fix it, just that it isn't a priority. Perhaps you should read https://www.techrepublic.com/article/why-open-source-developers-are-burning-out-no-respect/ because maybe then you will understand why we are not being unreasonable.

**Comment by KengoTODA** [comment]
Yes, your PR is welcome. [SpotBugsRunner](https://github.com/spotbugs/spotbugs-gradle-plugin/blob/3e27e36419f0a527553e4ff8fd6f186f1d17b8c4/src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsRunner.java) and its subclasses are good pointers to grasp how the gradle plugin runs SpotBugs.

Currently, we have three strategies to run SpotBugs, and none of them run SpotBugs as you mentioned. All of them run a new process or worker, because SpotBugs is Singleton and Gradle supports paralell build. JVM level isolation could be enough, but I haven't tried yet.

About output to stdout, I'm not sure why it is necessary for CI.

**Comment by xenoterracide** [comment]
You do realize though that a lot of us open source developers are already spread thin and have other things we're working on. We don't all have time to contribute to *your* project. Your project being any given project that is broken in some way.

I don't see an actual workaround, imho.
